extras/travis_build.sh

```diff
@@ -106,11 +106,16 @@ fi
 if [[ $OSTYPE = darwin* ]]; then
     export CC=$(ls /usr/local/bin/gcc-* | grep '^/usr/local/bin/gcc-\d$')
     export CXX=$(ls /usr/local/bin/g++-* | grep '^/usr/local/bin/g++-\d$')
-    CONFIGURE_BOOST="-Dwith-boost=OFF"
-else
-    CONFIGURE_BOOST="-Dwith-boost=ON"
 fi
- 
+
+# Download Boost
+wget --no-verbose https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.gz
+tar -xzf boost_1_72_0.tar.gz
+rm -fr boost_1_72_0.tar.gz
+cp -fr boost_1_72_0 $HOME/.cache
+rm -fr boost_1_72_0
+CONFIGURE_BOOST="-Dwith-boost=$HOME/.cache/boost_1_72_0"
+
```

Comment:
The above handles the boost library differently than, for example, music or sionlib. Is there a specific reason for this? If not, wouldn't it be better to implement this the same way? I would suggest to hide the above in a file extras/install_boost.sh, adjust .travis.yaml accordingly and wrap boost with an environment variable, same as it is for all other libraries.

Reply:
@gtrensch The idea was to always compile with Boost, like Travis currently does. But I can implement it in a way similar to the other libraries. Should some Travis jobs not compile with Boost?

Reply:
@hakonsbm The static code check (the first build configuration in the travis build matrix) does not require libboost. I will make a suggestion for a change and create a PR against your branch.

Reply:
@hakonsbm I have created the PR, please check.